src/lake/Lake/Load/Main.lean

```diff
@@ -189,7 +189,18 @@ def Workspace.updateAndMaterialize
           -- Load the package
           let depPkg ← liftM <| loadDepPackage dep leanOpts reconfigure
           if depPkg.name ≠ dep.name then
-            logWarning s!"{pkg.name}: package '{depPkg.name}' was required as '{dep.name}'"
+            if dep.name = .mkSimple "std" ∧ depPkg.name = .mkSimple "boost" then
+              logError "what was the 'std' package has been renamed to 'boost'; \
+                users should switch packages depending on new versions of std/boost \
+                to 'require boost' instead of 'require std' and also update the \
+                GitHub URL accordingly"
+            if !Platform.isWindows then
+              IO.FS.removeDirAll depPkg.dir -- cleanup
+            else
+              -- Deleting git repositories via IO.FS.removeDirAll does not work reliably on Windows
```

Comment:
I assume this is because the vscode process is holding onto a file handle somewhere? I would expect this to work fine from the command line if vscode is not open

Reply:
@eric-wieser The removal works from the command (e.g., with `rm`), so I don't think that is the problem. I believe the problem is that Lean uses `std::remove` in the C++ code for `removeFile`, which doesn't play nice with Windows. Rust [instead](https://doc.rust-lang.org/std/fs/fn.remove_file.html) uses `unlink` on Unix and `DeleteFile` on Windows

Reply:
A cursory google suggests that `std::remove` should largely work fine; the issue arises only if you try to delete the file while you (or something else) has a handle on the process.

Is there any reason not to call `IO.FS.removeDirAll` here anyway on the off-chance it does work, and catch the error on windows?